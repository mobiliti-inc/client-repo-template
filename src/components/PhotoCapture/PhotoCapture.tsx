import React, { Component, Fragment } from 'react';
import DialogModal from '../../components/Modal/DialogModal/DialogModal';
import PhotoRecBox from '../../components/PhotoRecBox/PhotoRecBox';
import TakePhotoModal from '../../components/AddLicenseModal/TakePhotoModal';
import { iconCamera } from '../../assets';

type PhotoCaptureProps = {
	modalIsVisible: boolean,
	onCameraAccessFail: (...args: any[]) => any,
	handleCapturedPhoto: (...args: any[]) => any,
	handleModalVisibility: (...args: any[]) => any,
	holderId?: string
};

type PhotoCaptureState = {
	isPhotoModalVisible: boolean,
	isTakePhotoModalVisible: boolean
};

class PhotoCapture extends Component<PhotoCaptureProps, PhotoCaptureState> {
	state = {
		isPhotoModalVisible: false,
		isTakePhotoModalVisible: false
	};
	static getDerivedStateFromProps(props, state) {
		if (props.modalIsVisible !== state.isPhotoModalVisible) {
			return {
				isPhotoModalVisible: props.modalIsVisible
			};
		}
		return null;
	}
	onUsePhoto = image => {
		this.handleTakePhotoModalVisibility();
		this.props.handleCapturedPhoto(image);
	};
	handlePhotoDialogModalVisibility = (isPhotoModalVisible = false) => {
		this.props.handleModalVisibility(isPhotoModalVisible);
		this.setState({
			isPhotoModalVisible
		});
	};
	handleTakePhotoModalVisibility = (isTakePhotoModalVisible = false) => {
		this.handlePhotoDialogModalVisibility();
		this.setState({
			isTakePhotoModalVisible
		});
	};
	render() {
		const { isPhotoModalVisible, isTakePhotoModalVisible } = this.state;
		const elementId = this.props.holderId || 'upload-image';
		return (
			<Fragment>
				<DialogModal
					modalIsVisible={isPhotoModalVisible}
					closeModal={() => this.handlePhotoDialogModalVisibility()}
					modalTitle="Select an option below to change your profile photo."
				>
					<Fragment>
						<div>
							<PhotoRecBox
								isVertical={false}
								boxTitle="Upload Photo"
								boxIcon={iconCamera}
								boxImage=""
								onBoxClick={() => document.getElementById(elementId).click()}
								onUploadImageInputChange={() => { }}
							/>
						</div>

						<div>
							<PhotoRecBox
								isVertical={false}
								boxTitle="Take photo with camera"
								boxIcon={iconCamera}
								boxImage=""
								onBoxClick={() => this.handleTakePhotoModalVisibility(true)}
								onUploadImageInputChange={() => { }}
							/>
						</div>
					</Fragment>
				</DialogModal>

				<TakePhotoModal
					modalIsVisible={isTakePhotoModalVisible}
					closeModal={() => this.handleTakePhotoModalVisibility()}
					onBack={() => {
						this.handleTakePhotoModalVisibility(false);
						return this.handlePhotoDialogModalVisibility(true);
					}}
					onUsePhoto={this.onUsePhoto}
					closeCamera={!isTakePhotoModalVisible}
					onCameraAccessFail={this.props.onCameraAccessFail}
					title="Take photo"
				/>
			</Fragment>
		);
	}
}

// PhotoCapture.defaultProps = {
// 	holderId: ''
// };

export default PhotoCapture;
