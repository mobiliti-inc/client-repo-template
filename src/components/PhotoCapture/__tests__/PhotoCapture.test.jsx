import PhotoCapture from '../PhotoCapture';

describe('PhotoCapture', () => {
	const initialProps = {
		modalIsVisible: false,
		onCameraAccessFail: jest.fn(),
		handleCapturedPhoto: jest.fn(),
		handleModalVisibility: jest.fn(),
	};

	const component = <PhotoCapture {...initialProps} />;
	it('should render', () => {
		expect(component).toMatchSnapshot();
	});

	it('should call handleUploadUserProfilePhoto and handleTakePhotoModalVisibility when onUsePhoto method is called', () => {
		const tree = shallow(component);
		tree.instance().handleTakePhotoModalVisibility = jest.fn();
		tree.instance().onUsePhoto('mock image');
		expect(tree.instance().handleTakePhotoModalVisibility).toHaveBeenCalled();
		expect(tree.instance().props.handleCapturedPhoto).toHaveBeenCalled();
	});

	it('should update isPhotoModalVisible correctly when handlePhotoDialogModalVisibility method is called', () => {
		const tree = shallow(component);
		expect(tree.state('isPhotoModalVisible')).toEqual(false);
		tree.setProps({ modalIsVisible: true });
		tree.instance().handlePhotoDialogModalVisibility(true);
		expect(tree.state('isPhotoModalVisible')).toEqual(true);
	});

	it('should update isTakePhotoModalVisible correctly when handleTakePhotoModalVisibility method is called', () => {
		const tree = shallow(component);
		expect(tree.instance().state.isTakePhotoModalVisible).toEqual(false);
		tree.instance().handleTakePhotoModalVisibility(true);
		expect(tree.instance().state.isTakePhotoModalVisible).toEqual(true);
	});
});
