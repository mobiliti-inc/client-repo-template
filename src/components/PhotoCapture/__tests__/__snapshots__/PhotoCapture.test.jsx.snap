// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoCapture should render 1`] = `
<PhotoCapture
  handleCapturedPhoto={[MockFunction]}
  handleModalVisibility={[MockFunction]}
  holderId=""
  modalIsVisible={false}
  onCameraAccessFail={[MockFunction]}
/>
`;
